web-app/src/main/webapp/script.js

```diff
@@ -61,6 +62,10 @@ window.onload = () => {
   document.getElementById('form-container').style.display = 'none';
   document.getElementById('report-button').addEventListener('click', showReportForm);
   const map = initMap();
+  const categories = Array.from(document.getElementsByClassName('category'));
+  const timeFrames = Array.from(document.getElementsByClassName('time-frame'));
+  categories.map(categoryElement => categoryElement.addEventListener('change', () => { loadPoliceReports(map) }));
+  timeFrames.map(timeFrameElement => timeFrameElement.addEventListener('change', () => { loadPoliceReports(map) }));
```

Comment:
nit: you can use https://javascript.info/event-delegation#:~:text=Introduction%20to%20Events-,Event%20delegation,handler%20on%20their%20common%20ancestor. for this sort of stuff.